packages/discovery/src/discovery.ts

```
@@ -125,9 +125,11 @@ export default class Discovery {
       locator += tag.toLowerCase();
     }
 
-    // if (await this.isLocatorUnique(locator, rootEl, true)) {
-    //   return locator;
-    // }
+    if (await this.isLocatorUnique(locator, rootEl, true)) {
+      return locator;
+    }
+
+    
```

There are still cases when the returned locator is not unique.
See this example:

config: 

 aliases: {
        info: [{ name: 'Legend', selectors: ['span', 'label'] }],
      },
      optimizer: {
        exclude: [
          {
            type: 'tag',
            value: ['legend', 'label', 'aside'],
          },
        ],
      },
    }
    
html:

<body>
        <label class="cls1" data-shrink="false" for="mui-59" id="mui-59-label">Dependency Name</label>
        <div class="cls2">
           <span>Port</span>
        </div>
        <fieldset aria-hidden="true" class="legend">
           <legend class="cls2">
              <span>Identifier</span>
           </legend>
        </fieldset>
        <label class="cls1" data-shrink="false" for="mui-60" id="mui-60-label">Identifier</label>
       </body>

In this case it returns:
div > span  - for the first span

and

span -> for the second span (this one is not unique)

at line 128 is returned 'fieldset' locator as unique, but our logic identifies only the direct children (eg: div > span)
fieldset > span is not a valid selector because span is not a direct child for fieldset and in this case 'fieldset > span' is not returned - which is ok. But in such cases, we should go deeper in the hierarchy and identify 'fieldset span' which is a valid selector and is unique.

I added a test for this (which currently fails - but we should fix the logic to pass this test)

---

This fix partially resolves the case. Before this fix, it returned: ['span', 'span'] - both are not unique. After this fix, it returns: ['div > span', 'span'] - only the first one is unique

---

And going further, in the above case, if we exclude the last tag from the selector, eg we exclude 'span' itself, the empty locator is returned. In these cases, we should throw an error / suggestive message.